clients/common/utility.cpp

```diff
@@ -60,6 +78,28 @@ int type2int<hipblasDoubleComplex>(hipblasDoubleComplex val)
 // Return path of this executable
 std::string hipblas_exepath()
 {
+#ifdef WIN32
+    std::vector<wchar_t> result(MAX_PATH + 1);
+    // Ensure result is large enough to accomodate the path
+    for(;;)
+    {
+        auto length = GetModuleFileNameW(nullptr, result.data(), result.size());
+        if(length < result.size() - 1)
+        {
+            result.resize(length);
+            result.shrink_to_fit();
+            break;
+        }
+        result.resize(result.size() * 2);
+    }
+
+    std::wstring wspath(result.data());
```

Comment:
The `std::wstring` constructor that takes a pointer and a size is typically more efficient than the one that depends on null termination. That said, I think it would be even better if you could replace your `result` variable with `wspath` to eliminate this copy entirely.

(There also seems to be a bug here. The value returned by `GetModuleFileNameW` is the length excluding the null terminator, so the `result.resize(length);` earlier in the function must drop the null terminator. The `wspath` constructor will therefore read past the end of `result`.)